zk_toolbox/crates/zk_inception/src/commands/prover/setup_keys.rs

```diff
@@ -0,0 +1,84 @@
+use anyhow::Ok;
+use common::{
+    check_prerequisites, cmd::Cmd, logger, spinner::Spinner, GCLOUD_PREREQUISITES,
+    GPU_PREREQUISITES,
+};
+use config::EcosystemConfig;
+use xshell::{cmd, Shell};
+
+use super::utils::get_link_to_prover;
+use crate::{
+    commands::prover::args::setup_keys::{Mode, Region, SetupKeysArgs},
+    messages::{MSG_GENERATING_SK_SPINNER, MSG_SK_GENERATED},
+};
+
+pub(crate) async fn run(args: SetupKeysArgs, shell: &Shell) -> anyhow::Result<()> {
+    let args = args.fill_values_with_prompt();
+
+    if args.mode == Mode::Generate {
+        check_prerequisites(shell, &GPU_PREREQUISITES, false);
+
+        let ecosystem_config = EcosystemConfig::from_file(shell)?;
+        let link_to_prover = get_link_to_prover(&ecosystem_config);
+        shell.change_dir(&link_to_prover);
+
+        let spinner = Spinner::new(MSG_GENERATING_SK_SPINNER);
+        let cmd = Cmd::new(cmd!(
+            shell,
+            "cargo run --features gpu --release --bin key_generator -- 
+            generate-sk-gpu all --recompute-if-missing
+            --setup-path=data/keys
+            --path={link_to_prover}/data/keys"
+        ));
+        cmd.run()?;
+        spinner.finish();
+        logger::outro(MSG_SK_GENERATED);
+    } else {
+        check_prerequisites(shell, &GCLOUD_PREREQUISITES, false);
+
+        let ecosystem_config = EcosystemConfig::from_file(shell)?;
+        let link_to_setup_keys = get_link_to_prover(&ecosystem_config).join("data/keys");
+        let path_to_keys_buckets =
+            get_link_to_prover(&ecosystem_config).join("setup-data-gpu-keys.json");
+
+        let region = args.region.expect("Region is not provided");
+
+        let file = std::fs::File::open(path_to_keys_buckets)
```

Comment:
please use shell and not direct `fs`. 
The current dir is properly set only for shell